java/piranha/src/test/java/com/uber/piranha/CorePiranhaTest.java

```diff
@@ -500,6 +500,203 @@ public void testIgnoresPrefixMatchFlag() throws IOException {
+            "import org.mockito.stubbing.OngoingStubbing;",
+            "class MockitoTest {",
+            " private XPTest experimentation;",
+            " public void evaluate() {",
+            "  Answer ans = new Answer<Integer>() {\n"
+                + "      public Integer answer(InvocationOnMock invocation) throws Throwable {\n"
+                + "        return (Integer) invocation.getArguments()[0];\n"
+                + "      }};",
+            "  when(experimentation.isToggleDisabled(\"STALE_FLAG\")).thenReturn(false);",
+            "  when(experimentation.isToggleEnabled(\"STALE_FLAG\")).thenThrow(new RuntimeException());",
+            "  when(experimentation.isToggleDisabled(\"STALE_FLAG\")).thenCallRealMethod();",
+            "  when(experimentation.isToggleEnabled(\"STALE_FLAG\")).then(ans);",
+            "  boolean b1 = someWrapper(when(experimentation.isToggleDisabled(\"STALE_FLAG\")).thenCallRealMethod());",
+            "  boolean b2 = someWrapper(when(experimentation.isToggleDisabled(\"OTHER_FLAG\")).thenCallRealMethod());",
+            "  someWhenWrapper(when(experimentation.isToggleDisabled(\"STALE_FLAG\"))).thenCallRealMethod();",
+            "  someWhenWrapper(when(experimentation.isToggleDisabled(\"OTHER_FLAG\"))).thenCallRealMethod();",
+            "  when(experimentation.isToggleEnabled(\"OTHER_FLAG\")).thenReturn(false);",
+            "  when(experimentation.isToggleEnabled(\"OTHER_FLAG\")).thenThrow(new RuntimeException());",
+            "  when(experimentation.isToggleDisabled(\"OTHER_FLAG\")).thenCallRealMethod();",
+            "  when(experimentation.isToggleEnabled(\"OTHER_FLAG\")).then(ans);",
+            "  when(foobar()).thenReturn(false);",
+            "  when(foobar()).thenAnswer(ans);",
+            " }",
+            " public boolean foobar() { return true;}",
+            " public boolean someWrapper(Object obj) { return true;}",
+            " public OngoingStubbing<Boolean> someWhenWrapper(OngoingStubbing<Boolean> x) { return x;}",
+            "}")
+        .addOutputLines(
+            "XPFlagCleanerSinglePositiveCase.java",
+            "package com.uber.piranha;",
+            "import static org.mockito.Mockito.when;",
+            "import org.mockito.invocation.InvocationOnMock;",
+            "import org.mockito.stubbing.Answer;",
+            "import org.mockito.stubbing.OngoingStubbing;",
+            "class MockitoTest {",
+            " private XPTest experimentation;",
+            " public void evaluate() {",
+            "  Answer ans = new Answer<Integer>() {\n"
+                + "      public Integer answer(InvocationOnMock invocation) throws Throwable {\n"
+                + "        return (Integer) invocation.getArguments()[0];\n"
+                + "      }};",
+            "",
+            "",
+            "",
+            "",
+            "",
+            "  boolean b1 = someWrapper(when(true).thenCallRealMethod());",
```

Comment:
Is this wrapping common?

Reply:
I don't know, TBH. I just added it as a corner case to the test suite, assuming that if such a scenario appears it should not be refactored because the method `someWrapper` could be producing some required side effect.

Reply:
no worries. this is fine. we can iterate on this subsequently.